rag/README.md

```diff
@@ -0,0 +1,100 @@
+# Retrieval Augmented Generation (RAG)
+
+This demo provides an example of using Retrieval Augmented Generation (RAG) to add additional context to an LLM chatbot. 
+
+## Build and Deploy Locally
+
+### Download LLM model(s)
+
+The two models that we have tested and recommend for this example are Llama2 and Mistral. The locations of the GGUF variants
+are listed below:
+
+* Llama2 - https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF/tree/main
+* Mistral - https://huggingface.co/TheBloke/Mistral-7B-Instruct-v0.1-GGUF/tree/main
+
+_For a full list of supported model variants, please see the "Supported models" section of the
+[llama.cpp repository](https://github.com/ggerganov/llama.cpp?tab=readme-ov-file#description)._
+
+This example assumes that the developer already has a copy of the model that they would like to use downloaded onto their host machine and located in the `/models` directory of this repo. 
+
+This can be accomplished with:
+
+```bash
+cd models
+wget https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF/resolve/main/llama-2-7b-chat.Q5_K_S.gguf
+cd ../
+```
+### Download the embedding model
+
+To encode our additional data and populate our vector database, we need an embedding model (a second language model) for this workflow. Here we will use `BAAI/bge-large-en-v1.5` all the necessary model files can be found and downloaded from https://huggingface.co/BAAI/bge-large-en-v1.5.
```

Comment:
So bge will always be the one to use or there could be different models? Does it depends on the primary model type we use? E.g this only works with llama2 and mistral

Reply:
No, there could be different models. But I wanted to keep this simple for now and not provide the additional options of choosing an embedding model as well. But maybe its better to make it another user option now?


It does not depend on the primary model. It only matters that what ever embedding model was used to create your vector database is the same one used at run time with the vector database.

Reply:
From a desktop perspective i thought that when users want to build/play with a rag sample, we make them pick a model, a rag and then proceed with the usual tasks (download, build, run, ....).

Reply:
yeah, there would just be an additional step so it would be like, pick a model, select rag recipe, (rag requires an embedding model) select an embedding model, then proceed as usual.